src/common/python.ts

```diff
@@ -12,28 +12,27 @@ export const onDidChangePythonInterpreter: Event<IInterpreterDetails> =
   onDidChangePythonInterpreterEvent.event;
 
 let _api: PythonExtension | undefined;
-async function getPythonExtensionAPI(): Promise<PythonExtension | undefined> {
-  if (_api) {
-    return _api;
-  }
-  _api = await PythonExtension.api();
-  return _api;
+export async function getPythonExtensionAPI(): Promise<PythonExtension> {
+  const api = _api || (await PythonExtension.api());
+  _api = api;
+  return api;
```

Comment:
Unrelated to your PR, we might have to make this optional again after #5

Reply:
Yes, we need to make this optional as part of #5